compact.go

```diff
@@ -633,21 +638,28 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		allSymbols  = make(map[string]struct{}, 1<<16)
 		closers     = []io.Closer{}
 		err         error
-		overlapping = false
-		lastMaxT    = int64(math.MinInt64)
+		maxtGlobal  = blocks[0].MaxTime()
+		mintGlobal  = blocks[0].MinTime()
+		overlapping bool
 	)
 	defer func() { closeAll(closers...) }()
 
 	for i, b := range blocks {
 		if !overlapping {
-			if b.MinTime() < lastMaxT {
-				c.metrics.overlappingBlocks.Inc()
-				level.Warn(c.logger).Log("msg", "found overlapping blocks during compaction", "ulid", meta.ULID)
+			if i > 0 && b.MinTime() < maxtGlobal && b.MaxTime() > mintGlobal {
```

Comment:
This logic can be wrong (if we are assuming that blocks are unsorted).
Example
```
time --->
block1: |-------|
block2:                             |-------|
block3:              |-------|
```
If we had the blocks in this order, block 3 satisfies this condition though no overlapping.

If we assume that blocks are sorted w.r.t. mint, we don't need to keep track of mint.